lib/cisco_node_utils/interface.rb

```diff
@@ -499,13 +475,11 @@ def default_mtu
     end
 
     def speed
-      config_get('interface', 'speed', name: @name)
+      val = config_get('interface', 'speed', name: @name)
+      val == 'auto' ? val : val.to_i
```

Comment:
Yes, `val == nil` is a problem that I thought I handled by adding `return val if val.nil?` but I guess I forgot to commit it.

It made sense to me to treat `speed` as an integer where applicable, but if you don't think it adds value I can get rid of it.  I also thought it would help in the puppet layer if we wanted to specify integers.